src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -352,7 +367,7 @@ public String getImageDownloadLink(long imageID) {
       BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
       ImageData image = browse.getImage(this.securityContext, imageID);
       if (image.getFormat() != null) {
-        downloadLinkAddress = "https://" + hostname + "/omero/webgateway/archived_files/download/"
+        downloadLinkAddress = "https://" + hostname + "/webgateway/archived_files/download/"
```

Comment:
You could define paths like "downloadPath" as private final fields of this class, so they are more easily interchangeable. Maybe you could also use Path path = Paths.get(hostname, downloadPath, imageID) to safely build part of this String, but I don't know if that the best idea, as this is a URL, which of course contains other special symbols/parts.

Reply:
Thanks for the suggestion, I am also looking into the idea of using linksuppliers as @Steffengreiner suggested. I am still thinking how to proceed, since the methods that create these url strings seem to not be doing much else, aside from checking that the connection is live and catching some exceptions. These methods are mostly dedicated to construct the string, so I wondering what type of refactor makes more sense.

I will think how to make it easier to refactor and safer to construct the url strings, I will try the "Paths" approach.

Reply:
I opened this issue to address this in a dedicated manner: https://github.com/qbicsoftware/omero-lib/issues/70